projects/storefrontlib/src/shared/components/item-counter/item-counter.component.ts

```diff
@@ -1,219 +1,118 @@
   /**
-   * If value is too small it will be set to min, if is too big it will be set to max.
+   * Holds the value of the counter, the state of the `FormControl`
+   * can be managed outside of the item counter.
    */
-  adjustValueInRange(incomingValue: number): number {
-    return incomingValue < this.min || !this.min
-      ? this.min
-      : incomingValue > this.max || !this.max
-      ? this.max
-      : incomingValue;
-  }
+  @Input() control: FormControl;
 
   /**
-   * Update model value and refresh input
+   * This can be used in case an item has a minmum order quantity.
+   * @default 1
    */
-  manualChange(newValue: number): void {
-    newValue = this.adjustValueInRange(newValue);
-    this.updateValue(newValue);
-    /* We use the value from the input, however, this value
-      is not the correct value that should be displayed. The correct value to display
-      is this.value, which the parent updates if the async call succeed. If the call
-      fails, then the input will need to display this.value, and not what the user
-      recently typed in */
-    this.renderer.setProperty(this.input.nativeElement, 'value', newValue);
-  }
-
-  onKeyDown(event: KeyboardEvent): void {
-    const handlers = {
-      ArrowDown: () => this.decrement(),
-      ArrowUp: () => this.increment(),
-    };
+  @Input() min = 1;
 
-    if (handlers[event.code]) {
-      handlers[event.code]();
-      event.preventDefault();
-      event.stopPropagation();
-    }
-  }
-
-  onBlur(event: FocusEvent): void {
-    this.focus = false;
-    event.preventDefault();
-    event.stopPropagation();
-    this.onTouch();
-  }
+  /**
+   * This can be used in case an item has a maximum order quantity.
+   */
+  @Input() max: number;
 
-  onFocus(event: FocusEvent): void {
-    this.focus = true;
-    event.preventDefault();
-    event.stopPropagation();
-    this.onTouch();
-  }
+  /**
+   * The step is used to increment the count. It is supposed to be a
+   * positive inteteger or float.
+   * @default 1
+   */
+  @Input() step = 1;
 
   /**
-   * Verify value that it can be incremented, if yes it does that.
+   * Inidicatas that the input can be manually set to zero,
+   * despite the fact that the input controls will be limited to
+   * the minimum. The zero value can be used to remove an item.
    */
-  increment(): void {
-    this.manualChange(this.value + this.step);
-    this.setFocus(true);
-  }
+  @Input() allowZero: false;
+
+  private _control$: Observable<FormControl>;
 
   /**
-   * Verify value that it can be decremented, if yes it does that.
+   * In readonly mode the item counter will only be shown as a label,
+   * the form controls are not rendered.
+   * Please not that readonly is different from the `disabled` form state.
+   * @default false
    */
-  decrement(): void {
-    this.manualChange(this.value - this.step);
-    this.setFocus(false);
-  }
+  @HostBinding('class.readonly') @Input() readonly = false;
```

Comment:
Readonly doesn't work as imagined. I could still change the value of the input.

Reply:
Good point. The `readonly` was introduced to avoid the plus/minus buttons, since we cannot drive this by the form disabled state. The form state should be driven by the  formControl. 
That being said, I understand the confusion and will have a 2nd look at this.

Reply:
Readonly wasn't properly working actually. I've implemented the feature and add a few tests around it. Please double check.